src/parser/shared/modules/HotTracker.js

```diff
@@ -87,37 +62,87 @@ class HotTracker extends Analyzer {
+      event.selectedCombatant = this.selectedCombatant;
+      hotDuration = this.hotInfo[spellId].durationConditions(event);
+    }
+
+    let maxDuration = -1;
+    if(this.hotInfo[spellId].maxDuration){
+      event.selectedCombatant = this.selectedCombatant;
+      maxDuration = this.hotInfo[spellId].maxDuration(event);
+    }
+
     const newHot = {
       start: event.timestamp,
-      end: event.timestamp + this.hotInfo[spellId].duration,
-      originalEnd: event.timestamp + this.hotInfo[spellId].duration,
+      end: event.timestamp + hotDuration,
+      originalEnd: event.timestamp + hotDuration,
       spellId, // stored extra here so I don't have to convert string to number like I would if I used its key in the object.
+      guid: this.currentGuid,
       name: event.ability.name, // stored for logging
       ticks: [], // listing of ticks w/ effective heal amount and timestamp, to be used as part of the HoT extension calculations
       attributions: [], // The effect or bonus that procced this HoT application. No attribution implies the spell was hardcast.
       extensions: [], // The effects or bonuses that caused this HoT to have extended duration. Format: { amount, attribution }
       boosts: [], // The effects or bonuses that caused the strength of this HoT to be boosted for its full duration.
+      healingAfterOriginalEnd: 0,//healing this hot did after the base duration due to extensions
+      maxDuration: maxDuration,//the true max duration that extensions will obey by (looking at rising mist)
     };
+
+    this.currentGuid += 1;
+
     if (!this.hots[targetId]) {
       this.hots[targetId] = {};
     }
     this.hots[targetId][spellId] = newHot;
+    this.hotHistory.push(newHot);
+  }
+
+  hotHeal(event) {
+    const spellId = event.ability.guid;
+    const target = this._getTarget(event);
+
+    if (!target) {
+      return;
+    }
+    const targetId = event.targetID;
+    const healing = event.amount + (event.absorbed || 0);
+
+    if (!this._validateHot(event)) {
+      return;
+    }
+    const hot = this.hots[targetId][spellId];
+    if (event.tick) { // direct healing (say from a PotA procced regrowth) still should be counted for attribution, but not part of tick tracking
+      hot.ticks.push({ healing, timestamp: event.timestamp });
+    }
 
-    if(this.selectedCombatant.hasBuff(SPELLS.THUNDER_FOCUS_TEA.id) && spellId === SPELLS.RENEWING_MIST_HEAL.id){
-      this.hots[targetId][spellId].end += TFT_REM;
+    if(hot.originalEnd < event.timestamp && event.timestamp < (hot.start + hot.maxDuration)){
+      hot.healingAfterOriginalEnd += healing;
     }
+
+    hot.attributions.forEach(att => {
+      att.healing += healing;
+    });
+    hot.boosts.forEach(att => {
+      att.healing += calculateEffectiveHealing(event, att.boost); 
+    });
+
+    if(event.hitType === HIT_TYPES.CRIT && hot.effectOnCrit){//if the heal crits and it does something fancy on crits do that 
```

Comment:
review on my own thing for memory. 
should be `this.hotInfo[spellId].effectOnCrit` on `hot.effectOnCrit`